pkg/k8s/services/crud.go

```diff
@@ -73,3 +73,25 @@ func DestroyDev(dev *model.Dev, c *kubernetes.Clientset) error {
 func Get(namespace, name string, c kubernetes.Interface) (*apiv1.Service, error) {
 	return c.CoreV1().Services(namespace).Get(name, metav1.GetOptions{})
 }
+
+//GetPortsByPod returns the ports exposed via endpoint of a given pod
+func GetPortsByPod(p *apiv1.Pod, c *kubernetes.Clientset) ([]int, error) {
+	eList, err := c.CoreV1().Endpoints(p.Namespace).List(metav1.ListOptions{})
```

Comment:
why endpoints instead of checking the pod directly?

Reply:
ports at the pod level are only decorative. the endpoints tell exactly which ports are mapped to a kubernetes service (healthy or not).